src/sql/parts/dashboard/tabs/dashboardInnerTab.contribution.ts

```diff
@@ -0,0 +1,111 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { IExtensionPointUser, ExtensionsRegistry } from 'vs/platform/extensions/common/extensionsRegistry';
+import { IJSONSchema } from 'vs/base/common/jsonSchema';
+import { localize } from 'vs/nls';
+import { join } from 'path';
+import { createCSSRule } from 'vs/base/browser/dom';
+import URI from 'vs/base/common/uri';
+
+import { registerInnerTab, generateInnerTabContentSchemaProperties } from 'sql/platform/dashboard/common/innerTabRegistry';
+
+export type IUserFriendlyIcon = string | { light: string; dark: string; };
+
+export interface IDashboardInnerTabContrib {
+	id: string;
+	title: string;
+	icon?: IUserFriendlyIcon;
+	content: object;
+}
+
+const innerTabSchema: IJSONSchema = {
+	type: 'object',
+	properties: {
+		id: {
+			type: 'string',
+			description: localize('sqlops.extension.contributes.dashboard.innertab.id', "Unique identifier for this inner tab. Will be passed to the extension for any requests.")
+		},
+		icon: {
+			description: localize('sqlops.extension.contributes.dashboard.innertab.icon', '(Optional) Icon which is used to represent this inner tab in the UI. Either a file path or a themable configuration'),
+			anyOf: [{
+				type: 'string'
+			},
+			{
+				type: 'object',
+				properties: {
+					light: {
+						description: localize('carbon.extension.contributes.account.icon.light', 'Icon path when a light theme is used'),
+						type: 'string'
+					},
+					dark: {
+						description: localize('carbon.extension.contributes.account.icon.dark', 'Icon path when a dark theme is used'),
+						type: 'string'
+					}
+				}
+			}]
+		},
+		title: {
+			type: 'string',
+			description: localize('sqlops.extension.contributes.dashboard.innertab.title', "Title of the inner tab to show the user.")
+		},
+		content: {
+			description: localize('sqlops.extension.contributes.dashboard.innertab.content', "The content that will be displayed in this inner tab."),
+			type: 'object',
+			properties: generateInnerTabContentSchemaProperties()
+		}
+	}
+};
+
+const innerTabContributionSchema: IJSONSchema = {
+	description: localize('sqlops.extension.contributes.innertabs', "Contributes a single or multiple inner tabs for users to add to their dashboard."),
+	oneOf: [
+		innerTabSchema,
+		{
+			type: 'array',
+			items: innerTabSchema
+		}
+	]
+};
+
+ExtensionsRegistry.registerExtensionPoint<IDashboardInnerTabContrib | IDashboardInnerTabContrib[]>('dashboard.innertabs', [], innerTabContributionSchema).setHandler(extensions => {
```

Comment:
You're really close on this design but I think are missing a key consideration: these "section groups" or inner tab contents are not special. It's a general concept that maps exactly to what Leila just added with grid / flow layout containers. We should merge the two and separate out the definition of a container & its contents from how it's composed into other things.

I think you should sync with Leila on this and agree on a standard way to do the following:
- Define a "Container" that can hold content. Current "container" types are:
  - A GridLayout or FlowLayout contribution
  - An insights section / contribution with multiple insights
  - A tasks section with multiple tasks
- Define how to compose a container into other parts. 
  - For top-level tabs, you can say "add this container as the tabs contents
  - For sections of this "Left Nav Container", you should call them "NavSections" rather than "InnerTabs" and define the object as "Icon + Title + Container"

This way we have a generic container with a bunch of content, that could be reused anywhere. It shouldn't matter that it's in a Nav view, inside a separate GridView, etc. it just "works" wherever you need it to.

Reply:
Will refactoring in another pr

---
In reply to: [169775213](https://github.com/Microsoft/sqlopsstudio/pull/751#discussion_r169775213) [](ancestors = 169775213)